### ISSUE TITLE
[BUG] When resampling pixel spacing

### ISSUE THREAD

**Issue by KathrynSch** [Radiomics/pyradiomics#544]
**Describe the bug**
Using the feature extractor on a 2D image with 2D mask raises a `ValueError` in the `resampleImage` of  `imageoperations.py`: 
`could not broadcast input array from shape (3) into shape (2)`.
This error is raised when the image and mask original pixel spacing equals to the `resampledPixelSpacing` defined in the feature extractor settings. 
The image and mask are of size (512, 512), mask contains integers [0, 1].

**PyRadiomics configuration**
The feature extractor has been initialized with the following settings:
```
settings = {
    'minimumROIDimensions': 2,
    'minimumROISize': None,
    'normalize': False,
    'normalizeScale': 1,
    'removeOutliers': None,
    'resampledPixelSpacing': [0.75, 0.75],
    'interpolator': 'sitkBSpline',
    'preCrop': False,
    'padDistance': 5,
    'distances': [1],
    'force2D': False,
    'force2Ddimension': 0,
    'resegmentRange': None,
    'label': 1,
    'additionalInfo': True}
``` 

**To Reproduce**
```
import SimpleITK as sitk
import radiomics
import numpy as np

# initialize feature extractor with default settings except resampledPixelSpacing
settings = {
    'minimumROIDimensions': 2,
    'minimumROISize': None,
    'normalize': False,
    'normalizeScale': 1,
    'removeOutliers': None,
    'resampledPixelSpacing': [0.7, 0.7],
    'interpolator': 'sitkBSpline',
    'preCrop': False,
    'padDistance': 5,
    'distances': [1],
    'force2D': False,
    'force2Ddimension': 0,
    'resegmentRange': None,
    'label': 1,
    'additionalInfo': True}
extractor = radiomics.featureextractor.RadiomicsFeatureExtractor(**settings)

# params to raise error
shape = (512, 512)
original_pixel_spacing = [0.7, 0.7]

# generate dummy data
image = np.random.rand(shape[0], shape[1])
mask = np.zeros(shape) # init mask with 0
mask[:10, :10] = np.ones((10, 10)) # simulate ROI

# convert image and mask to sitk images and set pixel spacing
image_sitk = sitk.GetImageFromArray(image)
image_sitk.SetSpacing(original_pixel_spacing)
mask_sitk = sitk.GetImageFromArray(mask)
mask_sitk.SetSpacing(original_pixel_spacing)

# extract radiomics
radiomic_features = extractor.execute(image_sitk, mask_sitk)
```

**Expected behavior**
The expected behavior is to receive an OrderedDic in return of the last line. 

**Version (please complete the following information):**
 - OS: Linux
 - Python version: 3.7
 - PyRadiomics version: 2.2.0

**Additional context**
I digged into the code a little bit and found the source of the problem. It is in the if case, line 477 of `imageoperations.py`:

```
# If current spacing is equal to resampledPixelSpacing, no interpolation is needed
# Tolerance = 1e-5 + 1e-8*abs(resampledSpacing)
logger.debug('Comparing resampled spacing to original spacing (image and mask')
if numpy.allclose(maskSpacing, resampledPixelSpacing) and numpy.allclose(imageSpacing, resampledPixelSpacing):
    logger.info('New spacing equal to old, no resampling required, applying pre-crop')
    low_up_bb = numpy.empty(Nd_mask * 2, dtype=int)
    low_up_bb[::2] = bb[:3]
    low_up_bb[1::2] = bb[:3] + bb[3:] - 1
    return cropToTumorMask(imageNode, maskNode, low_up_bb, **kwargs)

```

- `bb` in my case is an array of length 4 containing 2D bounding box coordinates (and not 6 in case of a 3D ROI)
- `low_up_bb` is of size `Nd_mask * 2` with `Nd_mask = 2` so 4.
- hence `low_up_bb[::2] = bb[:3]` raises a ValueError because the left-hand value is of size 2 and the right-hand value is of size 3. 

What should be done to fix this issue is to use `Nd_mask` as index instead of `3`:

```
low_up_bb[::2] = bb[:Nd_mask]
low_up_bb[1::2] = bb[:Nd_mask] + bb[Nd_mask:] - 1

```

**Comment by JoostJM** [comment]
Thanks for catching this! Do you want to make a pull request proposing this change?

**Comment by KathrynSch** [comment]
sure

**Comment by JoostJM** [comment]
Fixed by #545
